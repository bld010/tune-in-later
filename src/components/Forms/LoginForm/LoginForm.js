import React, { Component } from 'react';
import { loginUser } from '../../../util/apiCalls';
import './LoginForm.scss';

class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    } 
  }

  handleInputs = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  checkLoginStatus = (e) => {
    e.preventDefault();
    loginUser(this.state.email, this.state.password)
      .then(response => response.json())
      .then(data => console.log(data))
      .catch(error => console.error(error));
  }

  render() {
    // loginUser(this.state.email, this.state.password);
    
    return (
      <form className="login-form">
        <input className="email-input" placeholder="joanclarke@fempower.com" alt="email" name="email" value={this.state.email} onChange={this.handleInputs}></input>
        <input className="password-input" placeholder="type password here..." alt="password" name="password" value={this.state.password} onChange={e => this.handleInputs(e)}></input>
        <button className="login-btn" onClick={this.checkLoginStatus}>Login</button>
      </form>
    )
  }
} 

export default LoginForm;