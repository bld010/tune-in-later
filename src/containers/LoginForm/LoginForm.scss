.login-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

input {
  height: 30px;
  margin: 10px;
  padding: 5px;
  width: 25%;
}