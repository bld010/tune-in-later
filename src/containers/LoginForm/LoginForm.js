import React, { Component } from 'react';
import { loginUser, getUserFavoritesFromApi } from '../../util/apiCalls';
import { connect } from 'react-redux';
import { setCurrentUser, setCurrentUserFavorites } from '../../actions';
import PropTypes from 'prop-types';
import './LoginForm.scss';

export class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      error: ''
    } 
  }

  handleInputs = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleGetUserFavorites = (user) => {
    getUserFavoritesFromApi(user.id)
      .then(data => this.props.setCurrentUserFavorites(data.favorites))
  }

  checkLoginStatus = (e) => {
    e.preventDefault();
    loginUser(this.state.email, this.state.password)
      .then(data => {
        this.props.setCurrentUser(data)
        this.handleGetUserFavorites(data)
      })
      .catch(error => 
        this.setState({
          email: '',
          password: '',
          error: error.message
        })
      );
      this.clearLoginInputs();
  }
  
  clearLoginInputs = () => {
    this.setState({
      email: '',
      password: '',
      error: ''
    })
  }

  render() {
    // console.log(this.state.currentUser)
    return (
      <form className="login-form">
        <input className="email-input" placeholder="joanclarke@fempower.com" alt="email" name="email" value={this.state.email} onChange={this.handleInputs}></input>
        <input className="password-input" type="password" placeholder="type password here..." alt="password" name="password" value={this.state.password} onChange={e => this.handleInputs(e)}></input>
        {this.state.error && <p>Email and Password do not match.</p>}
        <button className="login-btn" onClick={this.checkLoginStatus}>Login</button>
      </form>
    )
  }
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser,
  favorites: state.favorites
})

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user)),
  setCurrentUserFavorites: favorites => dispatch(setCurrentUserFavorites(favorites))
})

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);

LoginForm.propTypes = {
  currentUser: PropTypes.object,
  favorites: PropTypes.array.isRequired,
  setCurrentUser: PropTypes.func.isRequired,
  setCurrentUserFavorites: PropTypes.func.isRequired
}