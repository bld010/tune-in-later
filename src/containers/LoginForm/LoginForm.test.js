import React from 'react'
import { LoginForm, mapStateToProps, mapDispatchToProps } from './LoginForm';
import { shallow } from 'enzyme';

describe('LoginForm', () => {
  let wrapper;

  const mockSetCurrentUser = jest.fn();
  const mockSetCurrentUserFavorites = jest.fn();
  const mockFavorites = [{
    id: 11,
    book_id: 918578041,
    user_id: 1,
    author_name: "Belmont and Belcourt Biographies",
    book_name: "Tim Tebow: An Unauthorized Biography (Unabridged)",
    artwork_url: "https://is1-ssl.mzstatic.com/image/thumb/Music1/v4/17/51/e6/1751e668-8249-cf42-5f3d-6e10c7398c1e/itunes.jpg/100x100bb.jpg",
    release_date: "2014-09-12T07:00:00Z",
    description: "Tim Tebow is one of the most talked about athletes of all time. Everybody seems to have an opinion about him, especially on his religious beliefs. But, in order to truly understand Tim Tebow, you must first understand where he came from. <i>Tim Tebow: An Unauthorized Biography</i> explores Tebow's upbringing, his high school and college football careers, and details his breakthrough into the NFL, all the way through his playoff loss to the New England Patriots. <br /><br />''Tebow-mania'' has fans and critics alike questioning his motives and looking for where God and his career will take him next. Inside, you will discover where it all started and learn why so many fans are ''Tebowing'' alongside their new national hero. From family to football to the future, all you need to know about Tim Tebow is right here, right now, in the most up to date coverage of Tim Tebow's life. <br /><br />This audiobook also includes all of Tim Tebow's college and professional football statistics.",
    primary_genre_name: "Biographies & Memoirs"
  }];
  const mockCurrentUser = {
    id: 1,
    name: "Joan",
    email: "exceeding@this.io"
  };
  
  beforeEach(() => {
    wrapper = shallow(<LoginForm 
      currentUser={ mockCurrentUser } 
      favorites={ mockFavorites } 
      setCurrentUser={ mockSetCurrentUser } 
      setCurrentUserFavorites={ mockSetCurrentUserFavorites } 
    />)

  });

    it('should match the snapshot when a user is logged in', () => {
      expect(wrapper).toMatchSnapshot();
    });

    // it('should return a current user')
});