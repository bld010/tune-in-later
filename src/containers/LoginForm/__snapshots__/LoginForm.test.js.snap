// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot when a user is logged in 1`] = `
<form
  className="login-form"
>
  <input
    alt="email"
    className="email-input"
    name="email"
    onChange={[Function]}
    placeholder="joanclarke@fempower.com"
    value=""
  />
  <input
    alt="password"
    className="password-input"
    name="password"
    onChange={[Function]}
    placeholder="type password here..."
    value=""
  />
  <button
    className="login-btn"
    onClick={[Function]}
  >
    Login
  </button>
</form>
`;
